### PR TITLE
UI for selecting preferred terminal

### LINKED ISSUE #112: New feature: Add UI for changing default terminal
This is an extension of #106 

The goal of it is to implement Picker UI to choose user's default terminal

### PR DESCRIPTION
**Pull request by gokul1099**
## Summary:
This pull request add a new feature to the preference tab, which allows user to change the preferred terminal of their choice.
closes #112

## Changelog:
[added] - added new picker for selecting terminal from the list of supported terminal

## Test Plan:
Screenshot 2023-12-18 at 9 27 44 PM

### CODE REVIEW

File: MiniSim/Views/Preferences.swift [hunk 1]
```diff
@@ -11,8 +11,30 @@ import Preferences
...
+           } else {
+               preferedTerminal = Terminal.terminal
+           }
+    }
+
     var body: some View {
         Settings.Container(contentWidth: 400) {
+            Settings.Section(title: "Preferred Terminal:") {
+                Picker("", selection: $preferedTerminal) {
+                    ForEach(Terminal.allCases, id: \.self) { terminal in
```
Comment (okwasniewski) [inline MiniSim/Views/Preferences.swift:29]: I think we need to check whether this app is installed on users device. Now he can pick an app that's not available

File: MiniSim/Views/Preferences.swift [hunk 2]
```diff
@@ -11,8 +11,30 @@ import Preferences
...
+                        Text(terminal.rawValue)
+                    }
+                }
+                .onChange(of: preferedTerminal) { _ in
+                    UserDefaults.standard.setValue(preferedTerminal.rawValue, forKey: UserDefaults.Keys.preferedTerminal)
```
Comment (okwasniewski) [inline MiniSim/Views/Preferences.swift:34]: Can you apply formatting here? SwiftLint added some warnings

File: MiniSim/Views/Preferences.swift (hunk 1 above)
Comment (gokuls1999) [reply MiniSim/Views/Preferences.swift:29]: Yes, We have to check, I missed that one. Will work on that

Comment (gokul1099) [reply MiniSim/Views/Preferences.swift:29]: Yes we need to check. I missed that one. Will work on that

**Review by okwasniewski — APPROVED** [review]
Thanks!

**Pull request merged**